netcore/System.Private.CoreLib/src/System.Runtime.Loader/AssemblyLoadContext.cs

```diff
@@ -87,5 +114,129 @@ internal static Assembly DoAssemblyResolve (string name)
 		{
 			return AssemblyResolve (null, new ResolveEventArgs (name));
 		}
+
+		// Invoked by Mono to resolve using the load method.
+		private static Assembly? MonoResolveUsingLoad (IntPtr gchALC, string assemblyName)
+		{
+			return Resolve (gchALC, new AssemblyName (assemblyName));
+		}
+
+		// Invoked by Mono to resolve using the Resolving event after
+		// trying the Load overried and TPA load context without
+		// success.
+		private static Assembly? MonoResolveUsingResolvingEvent (IntPtr gchALC, string assemblyName)
+		{
+			return ResolveUsingResolvingEvent (gchALC, new AssemblyName (assemblyName));
+		}
+
+
+#region Copied from AssemblyLoadContext.CoreCLR.cs
+		// WISH: let's share this code
+
+		// This method is invoked by the VM when using the host-provided assembly load context
+		// implementation.
+		private static Assembly? Resolve(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
+		{
+			AssemblyLoadContext context = (AssemblyLoadContext)(GCHandle.FromIntPtr(gchManagedAssemblyLoadContext).Target)!;
+
+			return context.ResolveUsingLoad(assemblyName);
+		}
+
+		// This method is invoked by the VM to resolve an assembly reference using the Resolving event
+		// after trying assembly resolution via Load override and TPA load context without success.
+		private static Assembly? ResolveUsingResolvingEvent(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
+		{
+			AssemblyLoadContext context = (AssemblyLoadContext)(GCHandle.FromIntPtr(gchManagedAssemblyLoadContext).Target)!;
+
+			// Invoke the AssemblyResolve event callbacks if wired up
+			return context.ResolveUsingEvent(assemblyName);
+		}
+
+		// This method is invoked by the VM to resolve a satellite assembly reference
+		// after trying assembly resolution via Load override without success.
+		private static Assembly? ResolveSatelliteAssembly(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
```

Comment:
This is being copied but is unused, though looking at CoreCLR it seems it should be? See https://github.com/dotnet/coreclr/blob/007c6bc5d749c6cbd5fd106a2053ef5563d12c1d/src/vm/appdomain.cpp#L6599-L6624 and https://github.com/dotnet/coreclr/blob/007c6bc5d749c6cbd5fd106a2053ef5563d12c1d/src/binder/clrprivbinderassemblyloadcontext.cpp#L61-L68

Reply:
Whoops, didn't grab all the code I meant to, updated links. Maybe chat at lunch about the satellite assembly stuff.

Reply:
Thanks for tracking this down. I admit I punted on satellite assembly stuff. I'll take a look at wiring it up